compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -1803,8 +1803,20 @@ public void exitSimpleLiteral(BallerinaParser.SimpleLiteralContext ctx) {
         TerminalNode node;
         DiagnosticPos pos = getCurrentPos(ctx);
         Set<Whitespace> ws = getWS(ctx);
-        if ((node = ctx.IntegerLiteral()) != null) {
-            this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.parseLong(getNodeValue(ctx, node)));
+
+        if (ctx.integerLiteral() != null) {
+            if ((node = ctx.integerLiteral().DecimalIntegerLiteral()) != null) {
+                this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.parseLong(getNodeValue(ctx, node)));
+            } else if ((node = ctx.integerLiteral().HexIntegerLiteral()) != null) {
+                this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.parseLong(getNodeValue(ctx, node)
```

Comment:
Shall we move this line out of the inner if-else block to reduce the code repetition?